kuka_rox_hardware_interface/src/rox_hardware_interface.cpp

```diff
@@ -252,8 +256,8 @@ return_type KukaRoXHardwareInterface::read(
     return return_type::OK;
   }
 
-  if (udp_replier_->ReceiveRequestOrTimeout(receive_timeout_) ==
-    UDPSocket::ErrorCode::kSuccess)
+  if (udp_replier_->ReceiveRequest() ==
```

Comment:
Keep in mind that this only works with a single hardware interface. If we're also adding a mobile platform, then the core control loop could easily get blocked if we aren't receiving a reply from the ECS running on the robot arm's KRC.

Reply:
@Svastits @VX792 I believe if this is resolved and everything tested again the PR should be ready.

Reply:
I have reverted to the original version, after testing it non-mock, we could merge this :)

Reply:
Original version with timeout is working